packages/node-resolve/src/package/resolvePackageImports.js

```diff
@@ -0,0 +1,49 @@
+import path from 'path';
+import fs from 'fs';
+import { pathToFileURL } from 'url';
+
+import { createBaseErrorMsg, InvalidModuleSpecifierError } from './utils';
+import resolvePackageImportsExports from './resolvePackageImportsExports';
+
+function findPackageJson(base) {
+  const { root } = path.parse(base);
+  let current = base;
+
+  while (current !== root && !current.endsWith('node_modules')) {
+    const pkgJsonPath = path.join(current, 'package.json');
+    if (fs.existsSync(pkgJsonPath)) {
```

Comment:
Could we use the async version and make the function async?

Reply:
Done